code/game/objects/items/explosives/grenades/marines.dm

```diff
@@ -218,18 +218,19 @@
 	qdel(src)
 
 
-/proc/flame_radius(radius = 1, turf/epicenter, burn_intensity = 25, burn_duration = 25, burn_damage = 25, fire_stacks = 15, int_var = 0.5, dur_var = 0.5, colour = "red") //~Art updated fire.
+/proc/flame_radius(radius = 1, turf/epicenter, burn_intensity = 25, burn_duration = 25, burn_damage = 25, fire_stacks = 15, colour = "red") //~Art updated fire.
 	if(!isturf(epicenter))
 		CRASH("flame_radius used without a valid turf parameter")
 	radius = clamp(radius, 1, 50) //Sanitize inputs
-	int_var = clamp(int_var, 0.1,0.5)
-	dur_var = clamp(int_var, 0.1,0.5)
-	fire_stacks = randfloat(burn_damage*(0.5-int_var),burn_damage*(0.5+int_var) ) + randfloat(burn_damage*(0.5-int_var),burn_damage*(0.5+int_var) )
-	burn_damage = randfloat(burn_damage*(0.5-int_var),burn_damage*(0.5+int_var) ) + randfloat(burn_damage*(0.5-int_var),burn_damage*(0.5+int_var) )
 
 	for(var/t in filled_turfs(epicenter, radius, "circle", air_pass = TRUE))
 		var/turf/turf_to_flame = t
-		turf_to_flame.ignite(randfloat(burn_intensity*(0.5-int_var), burn_intensity*(0.5+int_var)) + randfloat(burn_intensity*(0.5-int_var), burn_intensity*(0.5+int_var)), randfloat(burn_duration*(0.5-int_var), burn_duration*(0.5-int_var)) + randfloat(burn_duration*(0.5-int_var), burn_duration*(0.5-int_var)), colour, burn_damage, fire_stacks)
+		// Slight randomization on how long a fire last so that a sea of fire does not delete instantly at the same time. Flat for high durations, % for low.
+		if(burn_duration >= 10)
+			burn_duration = burn_duration - randfloat(0, 2)
+		else
+			burn_duration = randfloat(burn_duration, burn_duration*0.95)
```

Comment:
Keep the old duration variation of 50% or whatever. It looks far nicer, and is also far more interesting for gameplay as you get flamed areas that for a time can be traversed but are still hazardous.